sites/upsun/src/learn/overview/app-life-cycle.md

```diff
@@ -0,0 +1,193 @@
+
+This is the work that every instance must do to be ready for traffic, like preparing local caches, installing runtime files, or loading ML models.
+
+## Web commands
+
+Each web application on {{% vendor/name %}} can define web commands inside its `.upsun/config.yaml` file:
+
+```yaml
+applications:
+  myapp:
+    type: 'python:3.13'
+    web:
+      commands:
+        pre_start: "./scripts/setup.sh"
+        start: "uwsgi --ini conf/server.ini"
+        post_start: "./scripts/warm-up.sh"
+```
+
+| Name           | Type   | Required | Blocks Traffic        | Description                                                                                                                                               |
+| -------------- | ------ | -------- | --------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **pre_start**  | string | No       | No                    | Runs before `start`. Ideal for short per-instance setup tasks, such as preparing local caches or setting file permissions.                                |
+| **start**      | string | Yes      | Yes (until running)   | The main command that launches your app. Traffic is blocked until this command successfully runs. If it exits, {{% vendor/name %}} restarts the instance. |
+| **post_start** | string | No       | Yes (until completed) | Runs after the app starts but before the instance is added to the router. Use for warm-up tasks that must complete before serving traffic.                |
+
+For more information about web commands, visit the [Single-runtime Image page](/create-apps/app-reference/single-runtime-image.html#web-commands).
+
+{{< note theme="info" title="Blocking vs. non-blocking commands" >}}
+
+A command is blocking if an instance cannot receive traffic until it completes successfully.
+
+- `start` blocks traffic until the application is running.  
+- `post_start` blocks traffic until all warm-up tasks complete.  
+- `pre_start` is non-blocking.
```

Comment:
pre_start is also blocking, I'd remove the whole distinction of non-blocking/blocking, they all block the deployment.